### ISSUE TITLE
Entity reference revisions data producer returns null on empty

### ISSUE THREAD

**Issue by askibinski** [drupal-graphql/graphql#1151]
The entity_reference_revisions data producer currently returns null if no entites are referenced.
```
   * @return \GraphQL\Deferred|null
   *   A promise that will return entities or NULL if there aren't any.
```
My schema currently looks something like this:
```
type Page implements NodeInterface {
  id: Int!
  title: String!
  content: [ParagraphInterface]
}
```
Of course, frontend could also handle null but shouldn't we return an empty array here?
```
if (empty($entities)) {
  $type = $this->entityTypeManager->getDefinition($type);
  /** @var \Drupal\Core\Entity\EntityTypeInterface $type */
  $tags = $type->getListCacheTags();
  $context->addCacheTags($tags);
  return NULL;
}
```

edit: I realize the schema actually dictates this and we could always create own data producers, just wondering what about sensible defaults.

**Comment by joaogarin** [comment]
I agree I think an empty array makes more sense, if the data producer is for multiple entities anyway..it does return multiple entities if there are any, so an empty array if there are none makes sense

**Comment by hchonov** [comment]
An empty array is being returned by the EntityReference data producer as well. This is however a problem when being used in a composite resolver structure. I have a Count resolver running after the reference which is not being invoked because of the NULL value in the composition. So instead of returning 0 for the count my composition returns NULL and breaks a required integer value.

**Comment by klausi** [comment]
Fixed in #1265
